### PR TITLE
feat(ui): clicking on the user image shows profile modal

### PR DESCRIPTION
**Pull request by vimercati-samir**
What this PR does 📖
Clicking on the user image shows profile modal, if you click on your own profile SettingsModal shows up otherwise the generic user modal is opened

Which issue(s) this PR fixes 🔨
AP-508

Special notes for reviewers 🗒️

Additional comments 🎤

### CODE REVIEW

**Comment by stavares843** [comment]
@vimercati-samir please fill out our pretty template 🎨

File: components/views/navigation/mobile/nav/Nav.html [hunk 1]
```diff
@@ -1,27 +1,23 @@
...
   <div :class="`${$route.path.includes('/settings') ? 'active' : ''}`">
     <settings-icon
       size="1x"
-      v-on:click="$store.commit('ui/toggleSettings', true)" />
+      v-on:click="$store.commit('ui/toggleSettings', { show: true })"
+    />
   </div>
-</div>
\ No newline at end of file
+</div>
```
Comment (josephmcg) [inline components/views/navigation/mobile/nav/Nav.html]: I guess this is more of a preference, but I try to use the shorthand @click whenever I can. I think you can also remove the arrow functions in the click attributes

**Review by josephmcg — CHANGES_REQUESTED** [review]
Overall good, mostly cleanup old code from the same files you were updating.

One functionality question though.. Should clicking this open the profile modal?
It sets the active chat as well as opening the profile modal

image

File: components/views/settings/modal/Modal.vue [hunk 2]
```diff
@@ -18,13 +18,13 @@ export default Vue.extend({
...
-      page: 'personalize',
+      page: this.$store.state.ui.settingsDefaultRoute,
       settingSwiperOption: {
         initialSlide: 0,
         resistanceRatio: 0,
         slidesPerView: 'auto',
-        noSwiping: this.$device.isMobile ? false : true,
-        allowTouchMove: this.$device.isMobile ? true : false,
+        noSwiping: !this.$device.isMobile,
+        allowTouchMove: !!this.$device.isMobile,
```
[the remark this answers is not in the document]
Comment (vimercati-samir) [reply components/views/settings/modal/Modal.vue]: modified accordingly

File: components/views/navigation/mobile/nav/Nav.html (hunk 1 above)
Comment (vimercati-samir) [reply components/views/navigation/mobile/nav/Nav.html]: modified accordingly

File: store/ui/actions.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
+import Mousetrap from 'mousetrap'
 import { UIState } from './types'
 import { Channel } from '~/types/ui/server'
 import SoundManager, { Sounds } from '~/libraries/SoundManager/SoundManager'
 import { ActionsArguments } from '~/types/store/store'
 // @ts-ignore
```
[the remark this answers is not in the document]
Comment (vimercati-samir) [reply store/ui/actions.ts:6]: activateKeybinds and clearKeybinds depend on it, are you sure Mousetrap should be removed?

[the remark this answers is not in the document]
Comment (josephmcg) [reply store/ui/actions.ts:6]: I highlighted the wrong line by accident. I meant the ts-ignore can be removed. my bad lol
Thanks for confirming

**Comment by vimercati-samir** [comment]
> Overall good, mostly cleanup old code from the same files you were updating.
> 
> One functionality question though.. Should clicking this open the profile modal? It sets the active chat as well as opening the profile modal
> 
> image

Maybe instead of opening the profile modal when clicking on the user photo (user item in messages sidebar) the correct behaviour would be to open the chat
Screenshot 2022-01-20 at 11 39 39

Then clicking on the user photo in the chat header would open the profile modal
Screenshot 2022-01-20 at 11 43 11

What do you think about this flow?

**Comment by josephmcg** [comment]
I like that much better from a UX perspective

**Comment by stavares843** [comment]
thanks for adding the PR template info 💥

**Comment by WanderingHogan** [comment]
So we want to get to the profile when the user clicks the user image in the quick profile, not the list on the left. Sorry for the delay

**Comment by WanderingHogan** [comment]
image

**Comment by WanderingHogan** [comment]
1. clicking doesnt do anything
2. clicking opens quick profile
3. clicking opens full profile

**Comment by stavares843** [comment]
cc @vimercati-samir

**Comment by WanderingHogan** [comment]
@vimercati-samir closing this, please re-open when ready

**Pull request closed without merging**